src/redisearch_rs/trie_bencher/src/corpus.rs

```diff
@@ -1,75 +1,188 @@
-use std::{collections::BTreeSet, path::PathBuf};
+use std::{collections::BTreeSet, io::Cursor, path::PathBuf};
 
-use crate::RustTrieMap;
+use crate::bencher::rust_load_from_keys;
 
-/// If the corpus has already been downloaded, read it from disk.
-/// Otherwise, download it from the internet and save it to disk.
+/// This enum allows easy switching between different corpora for benchmarking.
 ///
-/// We also compute and save to disk a smaller corpus for microbenchmarks.
-pub fn download_or_read_corpus() -> String {
-    let data_dir = PathBuf::from("data");
-    let corpus_path = data_dir.join("1984.txt");
-    let corpus = if std::fs::exists(&corpus_path).ok() != Some(true) {
-        let corpus = download_corpus();
-        let checksum = crc32fast::hash(corpus.as_bytes());
-        assert_eq!(
-            checksum, 3817457071,
-            "The checksum of the downloaded corpus does not match the expected value. \
-            This may impact the accuracy and relevance of the operations being benchmarked. \
-            Review the diff before updating the expected checksum."
-        );
-        fs_err::write(&corpus_path, corpus.as_bytes()).expect("Failed to write corpus to disk");
+/// Users may call [CorpusType::download_or_read_corpus] to get the full content of the source files of a corpus or
+/// use the [CorpusType::create_keys] method that generates a [Vec<String>] containing the unique keys for trie construction. These
+/// must be further converted to the "legacy" string types: see [crate::strvec2_raw_words].
+///
+/// Information for remote files can be found in the folder: git_root/tests/benchmarks/
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum CorpusType {
+    /// uses a corpus from the redis benchmarks that contains 1k rows in a csv file describing wikipedia articles as a line
+    RedisBench1kWiki,
+
+    /// uses a corpus from the redis benchmarks that contains 10k rows in a csv file describing wikipedia articles as a line
+    RedisBench10kNumerics,
+
+    /// legacy corpus from gutenberg.net called 1984.txt
+    GutenbergEbook,
+}
+
+impl CorpusType {
+    /// If the corpus has already been downloaded, read it from disk.
+    /// Otherwise, download it from the internet and save it to disk.
+    /// In any case perform a crc32 check to notice changes in the corpus data.
+    pub fn download_or_read_corpus(&self) -> String {
+        let path = self.get_cached_path();
+        let corpus = if std::fs::exists(&path).ok() != Some(true) {
+            let corpus = download_corpus(self.get_url());
+            fs_err::write(&path, corpus.as_bytes()).expect("Failed to write corpus to disk");
```

Comment:
We need to make sure that the `data` directory exists, otherwise this write will fail.